service/matching/matchingEngine.go

```diff
@@ -399,10 +399,7 @@ pollLoop:
 				return emptyPollWorkflowTaskQueueResponse, nil
 			}
 
-			isStickyEnabled := false
-			if len(mutableStateResp.StickyTaskQueue.GetName()) != 0 {
-				isStickyEnabled = true
-			}
+			isStickyEnabled := taskQueueName == mutableStateResp.StickyTaskQueue.GetName()
```

Comment:
maybe just add a comment like
```suggestion
			// If a poll was redirected to a regular queue because there were no sticky workers
			// available, we might still see StickyTaskQueue set here for a regular poll. But we
			// should set this to false so that frontend returns full history.
			isStickyEnabled := taskQueueName == mutableStateResp.StickyTaskQueue.GetName()
```